conftest.py

```diff
@@ -46,3 +73,36 @@ def mock_lsf_version(monkeypatch, request):
     except OSError:
         # Provide a fake implementation of lsf_version()
         monkeypatch.setattr(dask_jobqueue.lsf, "lsf_version", lambda: "10")
+
+
+all_envs = {
+    None: LocalCluster,
+    "pbs": PBSCluster,
+    "moab": MoabCluster,
+    "slurm": SLURMCluster,
+    "sge": SGECluster,
+    "lsf": LSFCluster,
+    "oar": OARCluster,
+    "htcondor": HTCondorCluster,
+}
+
+
+@pytest.fixture
+def EnvSpecificCluster3(pytestconfig):
+    return all_envs[pytestconfig.getoption("-E")]
+
+
+@pytest.fixture(
+    params=[pytest.param(v, marks=[pytest.mark.env(k)]) for (k, v) in all_envs.items()]
+)
+def EnvSpecificCluster(request):
+    if request.param == HTCondorCluster:
+        dask.config.set({"jobqueue.htcondor.disk": "20MB"})
+    return request.param
+
+
+@pytest.fixture(params=list(all_envs.values()))
+def Cluster(request):
```

Comment:
So Cluster is all envs, but without specific CI?

Reply:
Yes. I thus runs as part of the default pytests without -E . I'll add a comment here to document this.